src/main/java/calculator/service/StringCalculate.java

```diff
@@ -0,0 +1,80 @@
+package calculator.service;
+
+import calculator.constant.Constant;
+import calculator.model.Calculator;
+import calculator.validator.InputValidator;
+
+import java.util.ArrayList;
+
+public class StringCalculate {
+
+    private static final String REGEX_START = "\\";
+    private static final String REGEX_OR = "|";
+
+    private String inputString;
+    Calculator calculator = Calculator.getInstance();
+
+    public StringCalculate(String inputString) {
+        this.inputString = inputString;
+    }
+
+    public void preprocessDelimiter() {
+        findCustomDelimiter();
+        takeOffCustomDelimiter();
+    }
+
+    private void findCustomDelimiter() {
+        if (inputString.startsWith(Constant.CUSTOM_DELIMITER_PREFIX)) {
+            char newDelimiter = inputString.charAt(2);
+            calculator.addCustomDelimiter(newDelimiter);
+        }
+    }
+
+    private void takeOffCustomDelimiter() {
+        if (inputString.startsWith(Constant.CUSTOM_DELIMITER_PREFIX)) {
+            inputString = inputString.substring(5);
+        }
+    }
+
+    public void splitAndSaveOperands() {
+        String[] stringOperands = splitOperands();
+        InputValidator.validateStringOperand(stringOperands);
+        ArrayList<Integer> intOperands = convertOperandsToInt(stringOperands);
+        saveOperands(intOperands);
+    }
+
+    private String[] splitOperands() {
+        String reg = "";
+        for (Character delimiter : calculator.getDelimiters()) {
+            reg += (REGEX_START + delimiter + REGEX_OR);
+        }
```

Comment:
String끼리의 연산은 비용이 많이든다는 것을 알고 계셨나요!?
https://dkswnkk.tistory.com/584

아래와 같이 작성하면 더 깔끔하고 효율적으로 작성할 수 있을 것 같습니다!
```suggestion
  String reg = calculator.getDelimiters().stream()
      .map(delimiter -> REGEX_START + delimiter)
      .collect(Collectors.joining(REGEX_OR));
```

Reply:
좋은 방법 알려주셔서 감사합니다! 스트림에 대해 배워봐야겠어요😀